lib/src/chart/line_chart/line_chart.dart

```diff
@@ -121,6 +185,23 @@ class _LineChartState extends AnimatedWidgetBaseState<LineChart> {
         ..clear()
         ..add(ShowingTooltipIndicators(sortedLineSpots));
     });
+
+    // If there is needed event and any lineBar with .isDraggable flag exists,
+    // sets indexes of needed spot and starts dragging process.
+    if (event is FlPanStartEvent || event is FlLongPressStart) {
+      final barIndex = touchResponse.lineBarSpots?.first.barIndex;
+      final spotIndex = touchResponse.lineBarSpots?.first.spotIndex;
+
+      if (spotIndex != null && barIndex != null) {
+        final isDraggable = widget.data.lineBarsData[barIndex].isDraggable;
```

Comment:
I'm a bit confused about the design, we have an `isDraggable` flag in our LineChartBarData class, also we have `DragSpotUpdateFinishedCallback? dragSpotUpdateFinishedCallback;` in our LineTouchData.
 (I know we might have something similar in other places, and I admit that is a bad design)

But for now, what if we keep these two variables beside each other in the LineTouchData class?
What do you think?

Reply:
Also, we can add other kinds of callbacks for example `dragSpotStartedCallback`, `dragSpotMovedCallback`, and `dragSpodFinishedCallback`. 

The other (perfect) solution is to touch the [RenderBaseChart](https://github.com/imaNNeo/fl_chart/blob/master/lib/src/chart/base/base_chart/render_base_chart.dart) to add a [DragGestureRecognizer](https://api.flutter.dev/flutter/gestures/DragGestureRecognizer-class.html) to detect the gesture events using this predefined, then we can pass down the gesture events to the touchCallback event as a `FlTouchEvent`. This way, we can use the `drag-drop` feature in all kinds of charts with the same logic.

For example in LineChart, we can have a flag to do the spot drag-drop (that you already implemented) as an under-the-hood magic. Also, users can implement any other kind of interaction with the `drag-drop` feature.

What do you think?

Reply:
Thanks for comments! I like the 2nd idea and I'll try to implement it asap :)

Reply:
> The other (perfect) solution is to touch the [RenderBaseChart](https://github.com/imaNNeo/fl_chart/blob/master/lib/src/chart/base/base_chart/render_base_chart.dart) to add a [DragGestureRecognizer](https://api.flutter.dev/flutter/gestures/DragGestureRecognizer-class.html) to detect the gesture events using this predefined, then we can pass down the gesture events to the touchCallback event as a `FlTouchEvent`. This way, we can use the `drag-drop` feature in all kinds of charts with the same logic.
> 
> For example in LineChart, we can have a flag to do the spot drag-drop (that you already implemented) as an under-the-hood magic. Also, users can implement any other kind of interaction with the `drag-drop` feature.
> 
> What do you think?

So, after some checks about the additional gesture recognizer, I realized, that it's already done :D

See, the [DragGestureRecognizer](https://api.flutter.dev/flutter/gestures/DragGestureRecognizer-class.html) is an abstract class.
```dart
abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
    // ...Its code her
}
```

However, the package already implements [PanGestureRecognizer](https://api.flutter.dev/flutter/gestures/PanGestureRecognizer-class.html) in the [RenderBaseChart](https://github.com/imaNNeo/fl_chart/blob/master/lib/src/chart/base/base_chart/render_base_chart.dart)  which extends [DragGestureRecognizer](https://api.flutter.dev/flutter/gestures/DragGestureRecognizer-class.html) and has everything we need to implement some kind of similar features in other charts as well.

Based on that, we already have the needed gesture recognizer and this PR just creates the wanted `under-the-hood magic.` for the `LineChart`.


> I'm a bit confused about the design, we have an `isDraggable` flag in our LineChartBarData class, also we have `DragSpotUpdateFinishedCallback? dragSpotUpdateFinishedCallback;` in our LineTouchData. (I know we might have something similar in other places, and I admit that is a bad design)
> 
> But for now, what if we keep these two variables beside each other in the LineTouchData class? What do you think?

To be honest, I don't really think what would be better here. My intention was to allow users to implement a draggable feature for particular `LineChartBarData`. For example, we can have a few bars, but only one of them is going to be draggable. It's similar to `isCurved` flag to me.

```dart
lineBarsData: [
    LineChartBarData(
        spots: spots1,
        isCurved: true,
        isDraggable: true,
    ),
    LineChartBarData(
        spots: spots2,
        isCurved: true,
        isDraggable: false,
        color: Colors.red,
    ),
],
```

So that's why I put this flag there.

> Also, we can add other kinds of callbacks for example `dragSpotStartedCallback`, `dragSpotMovedCallback`, and `dragSpodFinishedCallback`.
> 

Definitely! Added it in the commit f42ba9986be9f2833c1ed9b93cea4ef43cb63cba.
Anyway, we have to pay users' attention to be careful with using `setState` in the `dragSpotUpdateCallback`. Would be good to have some notes in the docs :)